src/link/edit.js

```diff
@@ -102,6 +108,59 @@ const Edit = ( {
 		}
 	};
 
+	const getFormatStart = ( position ) => {
+		if ( value.formats[ position ] && value.formats[ position ][ 0 ].type === formatType ) {
+			return getFormatStart( position - 1 );
+		}
+		return position;
+	};
+
+	const getFormatEnd = ( position ) => {
+		if ( value.formats[ position ] && value.formats[ position ][ 0 ].type === formatType ) {
+			return getFormatEnd( position + 1 );
+		}
+		return position;
+	};
+
+	const getTrimmedStart = ( selectedValue ) => {
+		const selectedString = selectedValue.text.slice( selectedValue.start, selectedValue.end );
+		const trimmed = selectedString.trimStart();
+		if ( selectedString.length !== trimmed.length ) {
+			const delta = selectedString.length - trimmed.length;
+			return selectedValue.start + delta;
+		}
+		return selectedValue.start;
+	};
+
+	const getTrimmedEnd = ( selectedValue ) => {
+		const selectedString = selectedValue.text.slice( selectedValue.start, selectedValue.end );
+		const trimmed = selectedString.trimEnd();
+		if ( selectedString.length !== trimmed.length ) {
+			const delta = selectedString.length - trimmed.length;
+			return selectedValue.end - delta;
+		}
+		return selectedValue.end;
+	};
+
+	const handleTextEdit = () => {
+		const differentLength = lastValue.formats.length !== value.formats.length;
+		const everyCharEqual = value.formats.every( ( f, i ) => lastValue.text[ i ] === value.text[ i ] );
+		const editDetected = differentLength || ! everyCharEqual;
```

Comment:
i assume you are checking length because checking every characters take more effort; 
if length has difference, then we don't need to calculate the characters difference.

Reply:
After further testing, looks like it's working better if we keep `differentLength` as it is (when a character is deleted, it's removed from `formats` and the `every` statement doesn't catch the difference). Will lookout for your thoughts after testing

Reply:
Why are you checking length + every char instead of standard string comparison? (str1 === str2)

Reply:
Wow yes that's much simpler, thanks